### ISSUE TITLE
This drawing disappears on Obsidian Mobile (both iOS and Android) when switching from markdown mode to Excalidraw mode

### ISSUE THREAD

**Issue by zsviczian** [zsviczian/obsidian-excalidraw-plugin#715]
[sample.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/9180915/sample.zip)

**Comment by zsviczian** [comment]
```markdown

---

excalidraw-plugin: parsed
tags: [excalidraw]
excalidraw-export-dark: false
excalidraw-link-prefix: ""

---

# Text Elements
It all started with a ^eTgwF3UK

BIG BANG ^OxBsGca6

NOTE ^dlZS5oOL

Every intellectual endeavour starts with a note.
Author:: [[Sönke Ahrens]] ^QQzgsJP4


# Embedded files
299140637906bfe2d3b43fa572073a56f4a6d8c5: [[Tools/Visual Templates/Icons/Icon - index-card, slip, note.svg]]

%%
# Drawing
```json
{
	"type": "excalidraw",
	"version": 2,
	"source": "https://excalidraw.com",
	"elements": [
		{
			"type": "rectangle",
			"version": 359,
			"versionNonce": 701487407,
			"isDeleted": false,
			"id": "H4yodQX9",
			"fillStyle": "solid",
			"strokeWidth": 1,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -189.87706122883378,
			"y": 221.57580661565385,
			"strokeColor": "transparent",
			"backgroundColor": "white",
			"width": 255,
			"height": 98,
			"seed": 21617,
			"groupIds": [],
			"strokeSharpness": "sharp",
			"boundElements": [
				{
					"type": "text",
					"id": "QQzgsJP4"
				}
			],
			"updated": 1658748192494,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 461,
			"versionNonce": 999173793,
			"isDeleted": false,
			"id": "6tlvlOD8wCCvNUzE7C_6S",
			"fillStyle": "hachure",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0.17699753499238113,
			"x": -94.5509452870444,
			"y": -233.3560909227698,
			"strokeColor": "black",
			"backgroundColor": "transparent",
			"width": 52.45005311629126,
			"height": 200.80557100523646,
			"seed": 1630345706,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 680,
			"versionNonce": 782911471,
			"isDeleted": false,
			"id": "ujbmbbmRPPYdRj1ysrp4-",
			"fillStyle": "hachure",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 2.2194765487953596,
			"x": -94.40086494224214,
			"y": -237.51771238844776,
			"strokeColor": "black",
			"backgroundColor": "transparent",
			"width": 51.39982059701607,
			"height": 208.52893314931086,
			"seed": 1304181802,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 605,
			"versionNonce": 502379137,
			"isDeleted": false,
			"id": "X_FG-zn9SJCYsAQ3XmHa9",
			"fillStyle": "hachure",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 1.0587208696118218,
			"x": -89.03302229365953,
			"y": -251.94514333234133,
			"strokeColor": "black",
			"backgroundColor": "transparent",
			"width": 45.656641129347655,
			"height": 237.58737153790514,
			"seed": 209961974,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 119,
			"versionNonce": 957968911,
			"isDeleted": false,
			"id": "-OHy_jDoOxkTQyGt0FxBB",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -3.2478872359910858,
			"y": -168.22783773060934,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 24.917157866208516,
			"height": 23.758220291036018,
			"seed": 653974006,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 270,
			"versionNonce": 351910497,
			"isDeleted": false,
			"id": "LE4tsS03vdqENWrdzQMec",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -110.84283783534731,
			"y": -74.16765024995273,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 24.917157866208516,
			"height": 23.758220291036018,
			"seed": 279946870,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 487,
			"versionNonce": 475945007,
			"isDeleted": false,
			"id": "vID23pDyaCiCvDm6Hz9G6",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 5.797510253542182,
			"x": -80.6362927686692,
			"y": -142.3795450417679,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 24.917157866208516,
			"height": 20.997599151311935,
			"seed": 1685979946,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "ellipse",
			"version": 347,
			"versionNonce": 561366593,
			"isDeleted": false,
			"id": "_r102jCQcD0E8dttrhVCr",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -118.18968818228335,
			"y": -198.56719568931004,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 24.917157866208516,
			"height": 23.758220291036018,
			"seed": 181657910,
			"groupIds": [
				"1vqyScXOCBY_j-FQLUlQt"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917082,
			"link": null,
			"locked": false
		},
		{
			"type": "text",
			"version": 556,
			"versionNonce": 778140239,
			"isDeleted": false,
			"id": "eTgwF3UK",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -174.2047017289857,
			"y": -20.73581568996684,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 216,
			"height": 27,
			"seed": 990863862,
			"groupIds": [
				"aBI8qa3YdI3QVbIxPrYxT"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917083,
			"link": null,
			"locked": false,
			"fontSize": 20,
			"fontFamily": 1,
			"text": "It all started with a",
			"rawText": "It all started with a",
			"baseline": 18,
			"textAlign": "center",
			"verticalAlign": "top",
			"containerId": null,
			"originalText": "It all started with a"
		},
		{
			"type": "text",
			"version": 594,
			"versionNonce": 2118473249,
			"isDeleted": false,
			"id": "OxBsGca6",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -163.7047017289857,
			"y": 6.941867340229237,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 195,
			"height": 47,
			"seed": 218554657,
			"groupIds": [
				"aBI8qa3YdI3QVbIxPrYxT"
			],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917083,
			"link": null,
			"locked": false,
			"fontSize": 36,
			"fontFamily": 1,
			"text": "BIG BANG",
			"rawText": "BIG BANG",
			"baseline": 32,
			"textAlign": "center",
			"verticalAlign": "top",
			"containerId": null,
			"originalText": "BIG BANG"
		},
		{
			"type": "line",
			"version": 316,
			"versionNonce": 1945852015,
			"isDeleted": false,
			"id": "rRFjq7npNI20Uyl3xvDyk",
			"fillStyle": "solid",
			"strokeWidth": 2,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -158.97595917717433,
			"y": 28.86341505348385,
			"strokeColor": "hsl(0, 80%, 50%)",
			"backgroundColor": "black",
			"width": 186.67138037975258,
			"height": 0,
			"seed": 1563445866,
			"groupIds": [],
			"strokeSharpness": "round",
			"boundElements": [],
			"updated": 1658745917083,
			"link": null,
			"locked": false,
			"startBinding": null,
			"endBinding": null,
			"lastCommittedPoint": null,
			"startArrowhead": null,
			"endArrowhead": null,
			"points": [
				[
					0,
					0
				],
				[
					186.67138037975258,
					0
				]
			]
		},
		{
			"type": "image",
			"version": 239,
			"versionNonce": 1582819841,
			"isDeleted": false,
			"id": "EU5uyjXEuiIEyZBydZP7O",
			"fillStyle": "hachure",
			"strokeWidth": 1,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -166.2047017289857,
			"y": 36.88509375811992,
			"strokeColor": "#000000",
			"backgroundColor": "transparent",
			"width": 200,
			"height": 140.98360655737704,
			"seed": 1179510689,
			"groupIds": [],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917083,
			"link": null,
			"locked": false,
			"status": "pending",
			"fileId": "299140637906bfe2d3b43fa572073a56f4a6d8c5",
			"scale": [
				1,
				1
			]
		},
		{
			"type": "text",
			"version": 273,
			"versionNonce": 1123316367,
			"isDeleted": false,
			"id": "dlZS5oOL",
			"fillStyle": "solid",
			"strokeWidth": 2,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -118.2047017289857,
			"y": 162.27760319559223,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 104,
			"height": 47,
			"seed": 1887031457,
			"groupIds": [],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658745917083,
			"link": null,
			"locked": false,
			"fontSize": 36,
			"fontFamily": 1,
			"text": "NOTE",
			"rawText": "NOTE",
			"baseline": 32,
			"textAlign": "center",
			"verticalAlign": "top",
			"containerId": null,
			"originalText": "NOTE"
		},
		{
			"type": "text",
			"version": 428,
			"versionNonce": 1341275871,
			"isDeleted": false,
			"id": "QQzgsJP4",
			"fillStyle": "solid",
			"strokeWidth": 4,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"angle": 0,
			"x": -184.87706122883378,
			"y": 226.57580661565385,
			"strokeColor": "black",
			"backgroundColor": "black",
			"width": 245,
			"height": 66,
			"seed": 467690497,
			"groupIds": [],
			"strokeSharpness": "sharp",
			"boundElements": [],
			"updated": 1658750908223,
			"link": "[[Sönke Ahrens]]",
			"locked": false,
			"fontSize": 16,
			"fontFamily": 1,
			"text": "Every intellectual endeavour\nstarts with a note.\nAuthor:: Sönke Ahrens",
			"rawText": "Every intellectual endeavour starts with a note.\nAuthor:: [[Sönke Ahrens]]",
			"baseline": 59,
			"textAlign": "left",
			"verticalAlign": "top",
			"containerId": "H4yodQX9",
			"originalText": "Every intellectual endeavour starts with a note.\nAuthor:: Sönke Ahrens"
		}
	],
	"appState": {
		"theme": "light",
		"viewBackgroundColor": "#ffffff",
		"currentItemStrokeColor": "#000000",
		"currentItemBackgroundColor": "transparent",
		"currentItemFillStyle": "hachure",
		"currentItemStrokeWidth": 1,
		"currentItemStrokeStyle": "solid",
		"currentItemRoughness": 1,
		"currentItemOpacity": 100,
		"currentItemFontFamily": 1,
		"currentItemFontSize": 20,
		"currentItemTextAlign": "left",
		"currentItemStrokeSharpness": "sharp",
		"currentItemStartArrowhead": null,
		"currentItemEndArrowhead": "arrow",
		"currentItemLinearStrokeSharpness": "round",
		"gridSize": null,
		"colorPalette": {}
	},
	"files": {}
}
```
%%

```

**Comment by zsviczian** [comment]
I am able to reproduce the exact same effect with the same file on Android.

**Comment by zsviczian** [comment]
apparently if the drawing has any embedded image, the error will happen.
I tried turning off compressed JSON - that did not help.

**Comment by zsviczian** [comment]
Resolved with a workaround in 1.7.9.
I was no longer able to reproduce the error with Obsidian Mobile 1.3.1 (64) on iOS, but was still experiencing the error on 1.3.1 (64) on Android. I was not able to reproduce the error at all on the desktop. Tried with 15.7 and 15.9.
